### PR TITLE
Feature/migrate modal styles to tail wind

### LINKED ISSUE #236: Migrate Modal component to tailwind
# Issue

## ZenHub Issue
The current Modal component is using SASS for styling, we need to remove all that code and use tailwind instead

### PR DESCRIPTION
**Pull request by tureey**
# Details
There are a few styles I don't know how to migrate:
`background-color` with opacity, `max-width`, `max-height`, `min-width` with values different than `0` or `100%`

## Description

Migrate the styles of the `Modal Component`

## Issue

https://github.com/Coding-Coach/coding-coach/issues/236

Closes #236

### CODE REVIEW

File: src/components/modal/modal.module.scss [hunk 1]
```diff
@@ -1,27 +1,11 @@
...
-    top: 0;
-    left: 0;
-    width:100%;
-    height: 100%;
-    background: rgba($color-black, .3);
-    display: flex;
-    align-items: center;
-    justify-content: center;
-}
+.bgColor { background: rgba($color-black, .3); }
```
Comment (crysfel) [inline src/components/modal/modal.module.scss]: We might want to create a plugin for this, to allow using this same styles somewhere else.

Comment (tureey) [reply src/components/modal/modal.module.scss]: Yes, I agree!

- Does it mean the rest of the PR is okay? *if that's the case,*
  - Does this line that you commented should block the PR?
  - Should we merge and create tech debt?

Comment (crysfel) [reply src/components/modal/modal.module.scss]: Yeah... I'm thinking in the plugin we should move all the styles related to the modal, and use the tailwind utilities there to set the responsive styles.

Comment (tureey) [reply src/components/modal/modal.module.scss]: I'll be waiting for news then! Okay? @crysfel

**Comment by tureey** [comment]
So, how we operate with the "plugin" thing? @crysfel

**Comment by tureey** [comment]
@crysfel I prefer to do in a new one

**Comment by tureey** [comment]
I added the modal background color in the palette. Maybe it's not the best practice. We'll have to add a component plugin maybe? @crysfel Let's close the branch if the rest of the things are OK and create another task to do it if that's okay

**Pull request merged**